### DISCUSSION TITLE
Loading official/custom plugins locally based on specific environment set

### DISCUSSION THREAD

**Discussion by hamzamihaidanielx** [parcel-bundler/parcel#5546] [Q&A] [answered]
So, basically, I checked the whole docs about the Plugins system, with transformers, optimizers, etc., and I was in a need for a plugin to obfuscate my final source code, which I already did locally and it works just fine, but I would like to have it loaded only on production environment only and seems like once that plugin has been set in .parcelrc config file it will always execute regardless the environment that's running on.

Is this even possible? I mean, even checked most of the source code, haha, and could not quite figure it out.

**Comment by mischnic** [comment, marked as answer]
So `options.mode === 'production'` doesn't work for you?

You could use an env variable 
https://github.com/parcel-bundler/parcel/blob/67aefd83f8a32813aa86c904952d6ceb63eb235c/packages/transformers/elm/src/ElmTransformer.js#L52
but for now the cache will not be invalidated so `--no-cache` would be needed when switching your "environment".

**Reply by hamzamihaidanielx** [reply]
> So `options.mode === 'production'` doesn't work for you?
> 
> You could use an env variable
> 
> https://github.com/parcel-bundler/parcel/blob/67aefd83f8a32813aa86c904952d6ceb63eb235c/packages/transformers/elm/src/ElmTransformer.js#L52
> 
> 
> but for now the cache will not be invalidated so `--no-cache` would be needed when switching your "environment".

OMG, YES!!!

It works perfectly now, haha, thank you a lot, and Happy Holidays!
